feat:适配了restClient,并撰写对应example

QHtttttt opened the pull request:
### Describe what this PR does / why we need it
适配了restClient,并撰写对应example
### Does this pull request fix one issue?
None

QHtttttt committed:
feat:适配了restClient,并撰写对应example

uuuyuqi commented on spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/restclient/SentinelRestClientBeanPostProcessor.java:
```diff
@@ -0,0 +1,44 @@
+		if (bean instanceof RestClient.Builder builder) {
+			builder.requestInterceptors(list -> {
+				if (!list.contains(interceptor)) {
+					list.add(interceptor);
```
1. 这里暂时这么写问题不大，不过 addFirst 更优。不过用户自己也利用 spring 一些扩展机制的情况下，first 也不一定能够保证了。 
2. 如果客户没有使用 spring bean 来管理，得在 readme 加一下你的示例

uuuyuqi commented on spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/restclient/SentinelRestClientInterceptor.java:
```diff
@@ -0,0 +1,114 @@
+		}
+
+		String resourceName = request.getMethod() + ":" + request.getURI();
+		System.out.println("[Sentinel-DEBUG] RestClient resourceName = " + resourceName);
```
要删除一下

uuuyuqi commented on spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/restclient/SentinelRestClientProperties.java:
```diff
@@ -0,0 +1,34 @@
+
+
+@ConfigurationProperties(prefix = "spring.cloud.sentinel.restclient")
+public class SentinelRestClientProperties {
```
长远来看，SentinelRestClientProperties 这个类是必要的，但是通过这里的 enable 值去控制 interceptor 编码行为并不是一种优雅的做法。

建议是 enable 为 false 的情况下，AutoConfiguration 下的那些 Sentinel bean 都不应该加载，这样不会对用户的使用有任何侵入。

你可以学习其他组件的做法，在 AutoConfiguration 类上加一个类似于@ConditionalOnProperty(name = "spring.cloud.sentinel.enabled", matchIfMissing = true) 这样的注解，来整体控制 Sentinel RestClient 组件加载。

QHtttttt committed:
根据PR意见修改优化：
1、用全局开关 spring.cloud.sentinel.enabled 统一 gating，移除运行时 enabled 判断；
2、拦截器插入位置改为 index 0（优先级更高）；
3、移除 System.out.println 调试输出；
4、README 新增 非 Spring 管理 RestClient 的手动集成示例。

uuuyuqi commented on spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/restclient/SentinelRestClientProperties.java:
```diff
@@ -0,0 +1,24 @@
+
+
+@ConfigurationProperties(prefix = "spring.cloud.sentinel.restclient")
+public class SentinelRestClientProperties {
```
没有使用的话，就删除吧，后续需要的情况下再加回来

uuuyuqi commented on spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/restclient/readme.md:
```diff
@@ -0,0 +1,110 @@
+
+  - 受 `spring.cloud.sentinel.enabled` 控制（`@ConditionalOnProperty`，`matchIfMissing=true`）。
+  - 声明两个 Bean：
+    - `SentinelRestClientInterceptor`（无参构造；不做 `System.out.println` 调试输出），
```
无参构造；不做 `System.out.println` 调试输出 --> 这句话是不是多余了
